File: C++/1190. Reverse Substrings Between Each Pair of Parentheses.cpp
class Solution {
public:
    string reverseParentheses(string s) {
        vector<int>a;
        for(int i=0;i<s.size();i++){
            if(s[i]=='(')
            a.push_back(i);
            else if(s[i]==')'){
                reverse(s.begin()+a.back(),s.begin()+i);
                a.pop_back();
            }
        }
        string k="";
        for(char c:s)
        if(c!=')'&&c!='(')k+=c;
        return k;
    }
};

File: JAVA/1190. Reverse Substrings Between Each Pair of Parentheses.java
class Solution {
    public String reverseParentheses(String s) {
        List<Integer> a = new ArrayList<>();
        StringBuilder sb = new StringBuilder(s);
        for (int i = 0; i < sb.length(); i++) {
            if (sb.charAt(i) == '(') {
                a.add(i);
            } else if (sb.charAt(i) == ')') {
                int x = a.remove(a.size() - 1);
                // Use StringBuilder reverse method
                String reversed = new StringBuilder(sb.substring(x + 1, i)).reverse().toString();
                sb.replace(x + 1, i, reversed);
            }
        }
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < sb.length(); i++) {
            char c = sb.charAt(i);
            if (c != '(' && c != ')') {
                result.append(c);
            }
        }
        return result.toString();
    }
}
